gis/bylaw_text_to_centreline/README.md

```diff
@@ -0,0 +1,300 @@
+
+    WHEN ST_LineLocatePoint(_wip2.whole_centreline, _wip2.oid1_geom)
+    > ST_LineLocatePoint(_wip2.whole_centreline, _wip2.oid1_geom_translated)
+    THEN ST_Difference(_wip2.pgrout_centreline ,
+    ST_LineSubstring(_wip2.pgrout_centreline, 
+    ST_LineLocatePoint(_wip2.pgrout_centreline, _wip2.oid1_geom) - (metres_btwn1/ST_Length(ST_Transform(_wip2.pgrout_centreline, 2952))),
+    ST_LineLocatePoint(_wip2.pgrout_centreline, _wip2.oid1_geom) )  
+        )
+
+    END )
+
+ELSE _wip2.pgrout_centreline
+
+END);
+```
+
+The image below shows the four possible cases (A1, A2, B1 and B2)
+![image](https://user-images.githubusercontent.com/54872846/77707649-284df380-6f9c-11ea-9509-b0db0cdf24c5.png)
+
+
+## Confidence output
+
+The function outputs a confidence level (`con`), which is defined after the `centreline_segments` variable. This value represents how close the names of the input values are to the intersections they were matched to. Specifically, this value is the sum of the levenshtien distance between the `highway2` value and the street name it matched to and the `btwn1`/`btwn2` value and the street name it matched to. It is the sum of this value for both intersections.
+
+# Quality Control
+
+There is a [Github Issue](https://github.com/CityofToronto/bdit_data-sources/issues/188) on weird bylaws that haven't been matched from Chelsea's work. \
+For example: \
+i) The streets stated in bylaws are slightly off i.e. they do not intersect or are rather far apart. \
+ii) Directionality matters. Northbound and Southbound route between two intersection are of different centrelines such as the Allen Rd. \
+iii) The error message `line_locate_point: 1st arg isn't a line` returned for 49 distinct id's. This mostly occurs when trying to slice a multilinestring. 
+
+Majority of them are also some issues that I have encountered with my new processes but have not been solved yet. I will list down the issue below and some idea on how we can tackle them.
+
+## pgRouting returns the shortest path but street name different from `highway`
+
+This can be found at [issue #268](https://github.com/CityofToronto/bdit_data-sources/issues/268#issuecomment-595973836). 
+This happened for `bylaw_id` = 6577 where `highway` = 'Garthdale Court' and `between` = 'Overbrook Place and Purdon Drive'. The two intersection ids found are 13448816 and 13448300 (marked as red cross below). The blue line is the result from pg_routing whereas the highlighted yellow path indicates the road segment from the bylaw.
+
+![image](https://user-images.githubusercontent.com/54872846/76124099-f7475800-5fc7-11ea-865e-963edefa43be.png)
+
+## Direction stated on bylaws is not taken into account
+
+Which is also somehow related to problem (ii) stated above. This can be found at [issue #276](https://github.com/CityofToronto/bdit_data-sources/issues/276). For example: `highway` = 'Black Creek Drive (southbound)' and `between` = 'Eglinton Avenue West and a point 200 metres north of Weston Road'
```

Comment:
I would just say these are the same issue and link them explicitly. The direction of the highway is mentioned in the bylaw but we don't use that and that's a problem. 

**Is there any info on how to identify if this is an issue for a set of processed bylaws?**